# Exploratory Data Analysis and Visualization of Films from 1980-2020

This project focuses on conducting exploratory data analysis and data visualization of films produced between 1980-2020. The project aims to analyze and explore the relationships between various features of the films, such as the genre, rating, budget, gross, director, writer, and runtime.

By conducting exploratory data analysis, we can identify trends, patterns, and relationships within the dataset, which can then be visualized using various data visualization techniques such as bar charts, scatter plots, and histograms. This will help to better understand the characteristics of the films produced during this period, and enable us to make predictions about future films.

## Imports and Reading Data

In [256]:
import pandas as pd
import numpy as np
import matplotlib.pylab as plt
import seaborn as sns
# Make float values look better
# We will use USA and UK format system
pd.options.display.float_format = '{:,.2f}'.format
sns.set_style('darkgrid')
sns.set_palette("Set2")

In [257]:
df = pd.read_csv('movies.csv')

## Data Understanding

- head()
- tail()
- info()
- Dataframe shape
- describe()
- isna()


In [258]:
df.head()

name rating      genre  year  \
0                                     The Shining      R      Drama  1980   
1                                 The Blue Lagoon      R  Adventure  1980   
2  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
3                                       Airplane!     PG     Comedy  1980   
4                                      Caddyshack      R     Comedy  1980   

                        released  score        votes         director  \
0  June 13, 1980 (United States)   8.40   927,000.00  Stanley Kubrick   
1   July 2, 1980 (United States)   5.80    65,000.00   Randal Kleiser   
2  June 20, 1980 (United States)   8.70 1,200,000.00   Irvin Kershner   
3   July 2, 1980 (United States)   7.70   221,000.00     Jim Abrahams   
4  July 25, 1980 (United States)   7.30   108,000.00     Harold Ramis   

                    writer            star         country        budget  \
0             Stephen King  Jack Nicholson  United Kingdom 19,000,000.00   
1  Henry De Vere Stacpoole  Brooke Shields   United States  4,500,000.00   
2           Leigh Brackett     Mark Hamill   United States 18,000,000.00   
3             Jim Abrahams     Robert Hays   United States  3,500,000.00   
4       Brian Doyle-Murray     Chevy Chase   United States  6,000,000.00   

           gross             company  runtime  
0  46,998,772.00        Warner Bros.   146.00  
1  58,853,106.00   Columbia Pictures   104.00  
2 538,375,067.00           Lucasfilm   124.00  
3  83,453,539.00  Paramount Pictures    88.00  
4  39,846,344.00      Orion Pictures    98.00

In [259]:
df.tail()

name rating   genre  year                          released  \
7663   More to Life    NaN   Drama  2020  October 23, 2020 (United States)   
7664    Dream Round    NaN  Comedy  2020  February 7, 2020 (United States)   
7665  Saving Mbango    NaN   Drama  2020         April 27, 2020 (Cameroon)   
7666   It's Just Us    NaN   Drama  2020   October 1, 2020 (United States)   
7667      Tee em el    NaN  Horror  2020   August 19, 2020 (United States)   

      score  votes       director         writer              star  \
7663   3.10  18.00  Joseph Ebanks  Joseph Ebanks      Shannon Bond   
7664   4.70  36.00   Dusty Dukatz    Lisa Huston  Michael Saquella   
7665   5.70  29.00   Nkanya Nkwai   Lynno Lovert      Onyama Laura   
7666    NaN    NaN  James Randall  James Randall     Christina Roz   
7667   5.70   7.00   Pereko Mosia   Pereko Mosia  Siyabonga Mabaso   

            country    budget  gross                    company  runtime  
7663  United States  7,000.00    NaN                        NaN    90.00  
7664  United States       NaN    NaN  Cactus Blue Entertainment    90.00  
7665  United States 58,750.00    NaN           Embi Productions      NaN  
7666  United States 15,000.00    NaN                        NaN   120.00  
7667   South Africa       NaN    NaN                PK 65 Films   102.00

In [260]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7668 entries, 0 to 7667
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   name      7668 non-null   object 
 1   rating    7591 non-null   object 
 2   genre     7668 non-null   object 
 3   year      7668 non-null   int64  
 4   released  7666 non-null   object 
 5   score     7665 non-null   float64
 6   votes     7665 non-null   float64
 7   director  7668 non-null   object 
 8   writer    7665 non-null   object 
 9   star      7667 non-null   object 
 10  country   7665 non-null   object 
 11  budget    5497 non-null   float64
 12  gross     7479 non-null   float64
 13  company   7651 non-null   object 
 14  runtime   7664 non-null   float64
dtypes: float64(5), int64(1), object(9)
memory usage: 898.7+ KB


In [261]:
df.describe()

year    score        votes         budget            gross  runtime
count 7,668.00 7,665.00     7,665.00       5,497.00         7,479.00 7,664.00
mean  2,000.41     6.39    88,108.50  35,589,876.19    78,500,541.02   107.26
std      11.15     0.97   163,323.76  41,457,296.60   165,725,124.32    18.58
min   1,980.00     1.90         7.00       3,000.00           309.00    55.00
25%   1,991.00     5.80     9,100.00  10,000,000.00     4,532,055.50    95.00
50%   2,000.00     6.50    33,000.00  20,500,000.00    20,205,757.00   104.00
75%   2,010.00     7.10    93,000.00  45,000,000.00    76,016,691.50   116.00
max   2,020.00     9.30 2,400,000.00 356,000,000.00 2,847,246,203.00   366.00

In [262]:
df.isna().sum()

name           0
rating        77
genre          0
year           0
released       2
score          3
votes          3
director       0
writer         3
star           1
country        3
budget      2171
gross        189
company       17
runtime        4
dtype: int64

In [263]:
# Converting the count of null values to a percentage
df.isna().sum() / df.shape[0] * 100

name        0.00
rating      1.00
genre       0.00
year        0.00
released    0.03
score       0.04
votes       0.04
director    0.00
writer      0.04
star        0.01
country     0.04
budget     28.31
gross       2.46
company     0.22
runtime     0.05
dtype: float64

## Data Preparation

- Identifying duplicated columns
- Deleting and filling some missing values

In [264]:
# Select duplicate rows in a DataFrame. (There is no duplicated rows)
df.loc[df.duplicated(subset = ['name','year','director'])]

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [265]:
df[df['budget'].isna()]

name     rating   genre  year  \
16                Fame          R   Drama  1980   
19          Stir Crazy          R  Comedy  1980   
24        Urban Cowboy         PG   Drama  1980   
25      Altered States          R  Horror  1980   
26     Little Darlings          R  Comedy  1980   
...                ...        ...     ...   ...   
7657              Leap        NaN   Drama  2020   
7659         I Am Fear  Not Rated  Horror  2020   
7660  Aloha Surf Hotel        NaN  Comedy  2020   
7664       Dream Round        NaN  Comedy  2020   
7667         Tee em el        NaN  Horror  2020   

                                released  score     votes            director  \
16          May 16, 1980 (United States)   6.60 21,000.00         Alan Parker   
19     December 12, 1980 (United States)   6.80 26,000.00      Sidney Poitier   
24          June 6, 1980 (United States)   6.40 14,000.00       James Bridges   
25     December 25, 1980 (United States)   6.90 33,000.00         Ken Russell   
26        March 21, 1980 (United States)   6.50  5,100.00         Ron Maxwell   
...                                  ...    ...       ...                 ...   
7657  September 25, 2020 (United States)   6.70    903.00   Peter Ho-Sun Chan   
7659       March 3, 2020 (United States)   3.40    447.00       Kevin Shulman   
7660    November 5, 2020 (United States)   7.10     14.00  Stefan C. Schaefer   
7664    February 7, 2020 (United States)   4.70     36.00        Dusty Dukatz   
7667     August 19, 2020 (United States)   5.70      7.00        Pereko Mosia   

                  writer              star        country  budget  \
16      Christopher Gore       Eddie Barth  United States     NaN   
19    Bruce Jay Friedman       Gene Wilder  United States     NaN   
24          Aaron Latham     John Travolta  United States     NaN   
25       Paddy Chayefsky      William Hurt  United States     NaN   
26             Kimi Peck      Tatum O'Neal  United States     NaN   
...                  ...               ...            ...     ...   
7657            Ji Zhang           Gong Li          China     NaN   
7659       Kevin Shulman    Kristina Klebe  United States     NaN   
7660  Stefan C. Schaefer       Augie Tulba  United States     NaN   
7664         Lisa Huston  Michael Saquella  United States     NaN   
7667        Pereko Mosia  Siyabonga Mabaso   South Africa     NaN   

              gross                                  company  runtime  
16    21,202,829.00                Metro-Goldwyn-Mayer (MGM)   134.00  
19   101,300,000.00                        Columbia Pictures   111.00  
24    46,918,287.00                       Paramount Pictures   132.00  
25    19,853,892.00                             Warner Bros.   102.00  
26    34,326,249.00  Stephen Friedman/Kings Road Productions    96.00  
...             ...                                      ...      ...  
7657  25,818,882.00                                      NaN   135.00  
7659      13,266.00                            Roxwell Films    87.00  
7660            NaN                      Abominable Pictures    90.00  
7664            NaN                Cactus Blue Entertainment    90.00  
7667            NaN                              PK 65 Films   102.00  

[2171 rows x 15 columns]

### Deleting Some Values (Rating / Gross)

- Missing rating values are 77 (1.004173 %)

- Missing gross values are 189 (2.464789 %)

In [266]:
df.dropna(subset = ['rating'],inplace = True)

In [267]:
df.dropna(subset = ['gross'], inplace = True)

In [268]:
df.isna().sum()

name           0
rating         0
genre          0
year           0
released       0
score          0
votes          0
director       0
writer         3
star           0
country        0
budget      2001
gross          0
company        9
runtime        1
dtype: int64

### Filling Missing Values

- director column
- budget column
- company column
- writer column
- runtime column
- released column

In [269]:
# I accidentally discovered these rows..
df[df['director'] == 'Directors']

name     rating      genre  year  \
117                        The Fox and the Hound          G  Animation  1981   
317    Bugs Bunny's 3rd Movie: 1001 Rabbit Tales          G  Animation  1982   
355                     Twilight Zone: The Movie         PG     Horror  1983   
773                                Godzilla 1985         PG     Action  1985   
783   He-Man and She-Ra: The Secret of the Sword          G  Animation  1985   
902                    The Great Mouse Detective          G  Animation  1986   
1011                   My Little Pony: The Movie          G  Animation  1986   
1103                    Amazon Women on the Moon          R     Comedy  1987   
1109                                        Aria          R     Comedy  1987   
2341                Batman: Mask of the Phantasm         PG  Animation  1993   
2357              We're Back! A Dinosaur's Story          G  Animation  1993   
2673                                  Four Rooms          R     Comedy  1995   
2913             Beavis and Butt-Head Do America      PG-13  Animation  1996   
3570                               Fantasia 2000          G  Animation  1999   
3834                          Digimon: The Movie         PG  Animation  2000   
4964                            Paris, je t'aime          R     Comedy  2006   
5071                                  Grindhouse          R     Action  2007   
5409                        New York, I Love You          R     Comedy  2008   
6144                                       V/H/S          R     Horror  2012   
6221                           The ABCs of Death  Not Rated     Comedy  2012   
6276                                    Movie 43          R     Comedy  2013   
6427                                     V/H/S/2          R     Horror  2013   
6817                                  Southbound          R     Horror  2015   
6854                                       Moana         PG  Animation  2016   
7001                                       Leap!         PG  Animation  2016   

                                released  score      votes   director  \
117        July 10, 1981 (United States)   7.30  87,000.00  Directors   
317    November 19, 1982 (United States)   7.10   1,800.00  Directors   
355        June 24, 1983 (United States)   6.50  34,000.00  Directors   
773      August 23, 1985 (United States)   6.20   6,100.00  Directors   
783       March 22, 1985 (United States)   7.30   2,600.00  Directors   
902         July 2, 1986 (United States)   7.20  46,000.00  Directors   
1011        June 6, 1986 (United States)   6.00   2,500.00  Directors   
1103  September 18, 1987 (United States)   6.20  10,000.00  Directors   
1109  September 15, 1987 (United States)   5.80   2,600.00  Directors   
2341   December 25, 1993 (United States)   7.80  45,000.00  Directors   
2357   November 24, 1993 (United States)   6.10  14,000.00  Directors   
2673   December 25, 1995 (United States)   6.80 100,000.00  Directors   
2913   December 20, 1996 (United States)   6.80  53,000.00  Directors   
3570       June 16, 2000 (United States)   7.20  36,000.00  Directors   
3834     October 6, 2000 (United States)   6.10   7,000.00  Directors   
4964       June 15, 2007 (United States)   7.20  71,000.00  Directors   
5071       April 6, 2007 (United States)   7.50 180,000.00  Directors   
5409    October 16, 2009 (United States)   6.20  45,000.00  Directors   
6144          September 6, 2012 (Russia)   5.80  58,000.00  Directors   
6221     April 26, 2013 (United Kingdom)   4.70  18,000.00  Directors   
6276    January 25, 2013 (United States)   4.30 101,000.00  Directors   
6427        June 6, 2013 (United States)   6.00  37,000.00  Directors   
6817    February 9, 2016 (United States)   5.90  17,000.00  Directors   
6854   November 23, 2016 (United States)   7.60 289,000.00  Directors   
7001     August 25, 2017 (United States)   6.80  21,000.00  Directors   

                     writer                     star         cou

In [270]:
# Found data from the internet
film_directors = {
    "The Fox and the Hound": "Ted Berman",
    "Bugs Bunny's 3rd Movie: 1001 Rabbit Tales": "Friz Freleng",
    "Twilight Zone: The Movie": "John Landis",
    "Godzilla 1985": "Koji Hashimoto",
    "He-Man and She-Ra: The Secret of the Sword": "Ed Friedman",
    "The Great Mouse Detective": "Ron Clements",
    "My Little Pony: The Movie": "Michael Joens",
    "Amazon Women on the Moon": "John Landis",
    "Aria": "Robert Altman",
    "Batman: Mask of the Phantasm": "Bruce Timm",
    "We're Back! A Dinosaur's Story": "Dick Zondag",
    "Four Rooms": "Allison Anders",
    "Beavis and Butt-Head Do America": "Mike Judge",
    "Fantasia 2000": "James Algar",
    "Digimon: The Movie": "Mamoru Hosoda",
    "Paris, je t'aime": "Olivier Assayas",
    "Grindhouse": "Robert Rodriguez",
    "New York, I Love You": "Fatih Akin",
    "V/H/S": "Matt Bettinelli-Olpin",
    "The ABCs of Death": "Kaare Andrews",
    "Movie 43": "Elizabeth Banks",
    "V/H/S/2": "Simon Barrett",
    "Southbound": "Roxanne Benjamin",
    "Moana": "Ron Clements",
    "Leap!": "Eric Summer"
}

for movie,director in film_directors.items():
    df.loc[df["name"] == movie, "director"] = director

In [271]:
df[df['director'] == 'Directors']

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [272]:
# filling missing budget values with mean of budgets
df['budget'].fillna(df['budget'].mean(),inplace = True)

In [273]:
# Checking if there is missing value left
df[df["budget"].isnull()]

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [274]:
df[df["company"].isnull()]

name   rating      genre  year  \
408    A Night in Heaven        R      Drama  1983   
633             The Bear       PG  Biography  1984   
969         Modern Girls    PG-13     Comedy  1986   
1033   P.O.W. the Escape        R     Action  1986   
1572      Heart of Dixie       PG      Drama  1989   
1594         Lost Angels        R      Drama  1989   
1630    Staying Together        R     Comedy  1989   
1806             Streets        R     Action  1990   
7599  End of the Century  Unrated      Drama  2019   

                                released  score    votes             director  \
408    November 18, 1983 (United States)   4.30 1,200.00     John G. Avildsen   
633   September 28, 1984 (United States)   6.10   270.00  Richard C. Sarafian   
969     November 7, 1986 (United States)   5.80 1,300.00         Jerry Kramer   
1033       April 4, 1986 (United States)   5.00   533.00          Gideon Amir   
1572     August 25, 1989 (United States)   5.20   677.00      Martin Davidson   
1594         May 5, 1989 (United States)   6.00   881.00          Hugh Hudson   
1630   November 10, 1989 (United States)   6.20   761.00            Lee Grant   
1806    January 19, 1990 (United States)   5.70   712.00            Katt Shea   
7599     August 16, 2019 (United States)   6.90 2,700.00         Lucio Castro   

                   writer                 star        country        budget  \
408       Joan Tewkesbury   Christopher Atkins  United States 35,998,144.20   
633          Michael Kane           Gary Busey  United States    221,000.00   
969          Laurie Craig        Daphne Zuniga  United States 35,998,144.20   
1033      Malcolm Barbour      David Carradine  United States 35,998,144.20   
1572  Anne Rivers Siddons          Ally Sheedy  United States  8,000,000.00   
1594       Michael Weller    Donald Sutherland  United States 35,998,144.20   
1630        Monte Merrick           Sean Astin  United States 35,998,144.20   
1806           Andy Ruben  Christina Applegate  United States 35,998,144.20   
7599         Lucio Castro       Juan Barberini      Argentina 35,998,144.20   

            gross company  runtime  
408  5,563,663.00     NaN    83.00  
633  2,687,148.00     NaN   110.00  
969    604,849.00     NaN    84.00  
1033 2,497,233.00     NaN    90.00  
1572 1,097,333.00     NaN    95.00  
1594 1,247,946.00     NaN   116.00  
1630 4,348,025.00     NaN    91.00  
1806 1,510,053.00     NaN    85.00  
7599   103,047.00     NaN    84.00

In [275]:
# Found data from the internet
movies_company_names = {
    'A Night in Heaven': '20th Century Fox',
    'The Bear': 'Renn Productions, Gaumont, Columbia Pictures',
    'Modern Girls': 'Atlantic Entertainment Group, New World Pictures',
    'P.O.W. the Escape': 'De Laurentiis Entertainment Group',
    'Heart of Dixie': 'HBO Pictures, Island Pictures',
    'Lost Angels': 'Hemdale Film Corporation, Island Pictures',
    'Staying Together': 'Hemdale Film Corporation',
    'Streets': 'Atlantic Entertainment Group, New World Pictures',
    'End of the Century': 'MAFILM, NFDC'
}

for names,company in movies_company_names.items():
    df.loc[df["name"] == names, "company"] = company

In [276]:
# Checking if there is missing value left
df[df["company"].isnull()]

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [277]:
df[df["writer"].isnull()]

name     rating      genre  year  \
1820             The Garden    Unrated      Drama  1990   
5834               The Trip  Not Rated     Comedy  2010   
7655  Legend of Deification      TV-PG  Animation  2020   

                             released  score     votes              director  \
1820             1990 (United States)   6.40    840.00          Derek Jarman   
5834          April 24, 2011 (Greece)   7.00 22,000.00  Michael Winterbottom   
7655  October 1, 2020 (United States)   6.60  1,300.00            Teng Cheng   

     writer            star         country        budget          gross  \
1820    NaN   Tilda Swinton  United Kingdom 35,998,144.20       5,006.00   
5834    NaN    Steve Coogan  United Kingdom 35,998,144.20   3,945,217.00   
7655    NaN  Guangtao Jiang           China 35,998,144.20 240,663,149.00   

                       company  runtime  
1820   Basilisk Communications    92.00  
5834      Baby Cow Productions   112.00  
7655  Beijing Enlight Pictures   110.00

In [278]:
df.loc[(df['name'] == "The Garden") & (df['director'] == 'Derek Jarman'),'writer'] = "Derek Jarman"
df.loc[(df['name'] == "The Trip") & (df['director'] == 'Michael Winterbottom'),'writer'] = "Steve Coogan"
df.loc[(df['name'] == "Legend of Deification") & (df['director'] == 'Teng Cheng'),'writer'] = "Li Wei"

In [279]:
# Checking if there is missing value left
df[df["writer"].isnull()]

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [280]:
df[df["runtime"].isnull()]

name rating   genre  year  \
6195  One for the Money  PG-13  Action  2012   

                              released  score     votes             director  \
6195  January 27, 2012 (United States)   5.30 41,000.00  Julie Anne Robinson   

             writer             star        country        budget  \
6195  Stacy Sherman  Katherine Heigl  United States 40,000,000.00   

             gross                  company  runtime  
6195 38,084,162.00  Lakeshore Entertainment      NaN

In [281]:
df.loc[(df['name'] == "One for the Money") & (df['director'] == 'Julie Anne Robinson'),'runtime'] = 91.0

In [282]:
# Checking if there is missing value left
df[df["runtime"].isnull()]

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [286]:
# There are no missing values anymore

df.isna().sum()

name        0
rating      0
genre       0
year        0
released    0
score       0
votes       0
director    0
writer      0
star        0
country     0
budget      0
gross       0
company     0
runtime     0
dtype: int64

In [287]:
# Initially there were 7668 pieces of data
# Now there are 7425 pieces
df.shape

(7425, 15)

### Editing Released Column

In [288]:
# Let's clean up the values in the "Released" column from unnecessary details 
# and convert the data type from object to datetime.

df["released"] = df["released"].astype(str)

df['released'] = df['released'].str.replace(r'\(.*\)', '',regex = True).str.strip()

# This code doesn't work right now because not all values are like 'June 13, 1984'
#df['released'] = pd.to_datetime(df['released'], format='%B %d, %Y')

# errors = 'coerce' generates NaT values instead of bad entries
df['released'] = pd.to_datetime(df['released'], format='%B %d, %Y', errors='coerce')

# Determining values that don't have %B %d, %Y format
na_values = df.loc[df['released'].isna()]
na_values

name     rating      genre  \
312                              Five Days One Summer         PG      Drama   
376                                        Nostalghia  Not Rated      Drama   
439                                     Heat and Dust          R      Drama   
449                                     Getting It on          R     Comedy   
463                                        Slayground          R      Crime   
467                              My Brother's Wedding  Not Rated      Drama   
719                                           Tampopo  Not Rated     Comedy   
731                                   Trouble in Mind          R     Comedy   
786                                      Taipei Story  Not Rated      Drama   
800                                   O.C. and Stiggs          R     Comedy   
1035                                     Duet for One          R      Drama   
1173                      The House on Carroll Street         PG   Thriller   
1212                                     Five Corners          R      Crime   
1227                                    A Man in Love          R      Drama   
1385                              Drowning by Numbers          R     Comedy   
1395                                         Chocolat      PG-13      Drama   
1404                                    Stormy Monday          R      Crime   
1415                                 Pascali's Island      PG-13      Drama   
1428                                   Platoon Leader          R      Drama   
1599                               Slaves of New York          R     Comedy   
1602  Scenes from the Class Struggle in Beverly Hills          R     Comedy   
1621                                    Monsieur Hire      PG-13      Crime   
1623                            Too Beautiful for You          R     Comedy   
1636                            Rosalie Goes Shopping         PG     Comedy   
1680                                           Nikita          R     Action   
1735                         The Comfort of Strangers          R      Drama   
1753                                    Life Is Sweet          R     Comedy   
1765                                        The Field      PG-13      Drama   
1772                                           Ju Dou      PG-13      Drama   
1797                                    Hidden Agenda          R      Drama   
1804                                My Father's Glory          G  Adventure   
1819                            Come See the Paradise          R      Drama   
1820                                       The Garden    Unrated      Drama   
1823                                        May Fools          R     Comedy   
1841                                   The Nasty Girl      PG-13     Comedy   
1947                                     Hear My Song          R     Comedy   
1985                                     The Adjuster          R     Comedy   
2000                                        Impromptu      PG-13  Biography   
2029                                      Liebestraum          R    Mystery   
2318                                      Killing Zoe          R      Crime   
2319                                           Cronos          R    Fantasy   
2427                               Bhaji on the Beach          R     Comedy   
2525                                          Torment  Not Rated      Crime   
2816                                The White Balloon    Unrated      Drama   
4187                                           Dahmer          R  Biography   
5833                                       Hatchet II          R     Action   
5958           The Human Centipede II (Full Sequence)  Not Rated     Horror   
6413                                 Romeo and Juliet      PG-13      Drama   

      year released  score     votes              director  \
312   1982      NaT   6.10  1,000.00        Fred Zinnemann   
376   1983      NaT   8.10 24,000.00      Andrei Tark

In [289]:
# Found data from the internet
# Changing some NaT values

movies_and_dates = {
  "Five Days One Summer": "1982-07-23",
  "Nostalghia": "1983-06-01",
  "Heat and Dust": "1983-08-05",
  "Getting It on": "1983-11-04",
  "Slayground": "1984-01-13",
  "My Brother's Wedding": "1984-09-07",
  "Tampopo": "1985-04-19",
  "Trouble in Mind": "1985-09-13",
  "Taipei Story": "1985-11-20",
  "O.C. and Stiggs": "1987-03-27",
  "Duet for One": "1988-05-27",
  "The House on Carroll Street": "1988-08-05",
  "Five Corners": "1988-09-09",
  "A Man in Love": "1988-09-23",
  "Drowning by Numbers": "1988-10-14",
  "Chocolat": "1988-12-22",
  "Stormy Monday": "1988-12-23",
  "Pascali's Island": "1988-12-23",
  "Platoon Leader": "1988-12-31",
  "Slaves of New York": "1989-03-17",
  "Scenes from the Class Struggle in Beverly Hills": "1989-04-05",
  "Monsieur Hire": "1989-05-03",
  "Too Beautiful for You": "1989-05-24",
  "Rosalie Goes Shopping": "1989-08-11",
  "Nikita": "1990-02-21",
  "The Comfort of Strangers": "1991-06-14",
  "Life Is Sweet": "1991-10-25",
  "The Field": "1991-12-20",
  "Ju Dou": "1991-12-25",
  "Hidden Agenda": "1991-12-25",
  "My Father's Glory": "1991-12-25",
  "Come See the Paradise": "1991-12-25",
  "The Garden": "1991-12-25",
  "May Fools": "1991-12-25",
  "The Nasty Girl": "1991-12-25",
  "Hear My Song": "1991-12-27",
  "The Adjuster": "1992-05-15",
  "Impromptu": "1992-06-05",
  "Liebestraum": "1992-08-14",
  "Killing Zoe": "1994-10-01",
  "Cronos": "1994-10-07",
  "Bhaji on the Beach": "1994-11-10",
  "Torment": "1994-12-02",
  "The White Balloon": "1995-02-17",
  "Dahmer": "2002-06-21",
  "Hatchet II": "2010-08-26",
  "The Human Centipede II (Full Sequence)": "2011-10-07",
  "Romeo and Juliet": "2013-10-11"
}

for names, dates in movies_and_dates.items():
    df.loc[df["name"] == names,"released"] = pd.to_datetime(dates, format = '%Y-%m-%d')

In [290]:
df.loc[df['released'].isna()]

Empty DataFrame
Columns: [name, rating, genre, year, released, score, votes, director, writer, star, country, budget, gross, company, runtime]
Index: []

In [291]:
# Let's remove year column, we will use released column when we need year value

df = df.drop(['year'], axis = 1)

In [292]:
df.head()

name rating      genre  \
0                                     The Shining      R      Drama   
1                                 The Blue Lagoon      R  Adventure   
2  Star Wars: Episode V - The Empire Strikes Back     PG     Action   
3                                       Airplane!     PG     Comedy   
4                                      Caddyshack      R     Comedy   

    released  score        votes         director                   writer  \
0 1980-06-13   8.40   927,000.00  Stanley Kubrick             Stephen King   
1 1980-07-02   5.80    65,000.00   Randal Kleiser  Henry De Vere Stacpoole   
2 1980-06-20   8.70 1,200,000.00   Irvin Kershner           Leigh Brackett   
3 1980-07-02   7.70   221,000.00     Jim Abrahams             Jim Abrahams   
4 1980-07-25   7.30   108,000.00     Harold Ramis       Brian Doyle-Murray   

             star         country        budget          gross  \
0  Jack Nicholson  United Kingdom 19,000,000.00  46,998,772.00   
1  Brooke Shields   United States  4,500,000.00  58,853,106.00   
2     Mark Hamill   United States 18,000,000.00 538,375,067.00   
3     Robert Hays   United States  3,500,000.00  83,453,539.00   
4     Chevy Chase   United States  6,000,000.00  39,846,344.00   

              company  runtime  
0        Warner Bros.   146.00  
1   Columbia Pictures   104.00  
2           Lucasfilm   124.00  
3  Paramount Pictures    88.00  
4      Orion Pictures    98.00

In [293]:
df.tail()

name     rating      genre   released  score    votes  \
7652      The Eight Hundred  Not Rated     Action 2020-08-28   6.80 3,700.00   
7653             The Quarry          R      Crime 2020-04-17   5.40 2,400.00   
7655  Legend of Deification      TV-PG  Animation 2020-10-01   6.60 1,300.00   
7656                  Tulsa      PG-13     Comedy 2020-06-03   5.00   294.00   
7659              I Am Fear  Not Rated     Horror 2020-03-03   3.40   447.00   

           director         writer             star        country  \
7652        Hu Guan        Hu Guan  Zhi-zhong Huang          China   
7653    Scott Teems    Scott Teems     Shea Whigham  United States   
7655     Teng Cheng         Li Wei   Guangtao Jiang          China   
7656    Scott Pryor    Scott Pryor      Scott Pryor  United States   
7659  Kevin Shulman  Kevin Shulman   Kristina Klebe  United States   

            budget          gross                              company  \
7652 80,000,000.00 461,421,559.00  Beijing Diqi Yinxiang Entertainment   
7653 35,998,144.20       3,661.00                     Prowess Pictures   
7655 35,998,144.20 240,663,149.00             Beijing Enlight Pictures   
7656 35,998,144.20     413,378.00                  Pryor Entertainment   
7659 35,998,144.20      13,266.00                        Roxwell Films   

      runtime  
7652   149.00  
7653    98.00  
7655   110.00  
7656   120.00  
7659    87.00